src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -205,22 +213,45 @@ def _clear_ready_queue(self) -> int:
     def join(self, timeout: Optional[float] = None) -> None:
         deadline = time.time() + timeout if timeout else None
         self._bulk_commit()
```

Comment:
Should we call `mark_commit_ready` before this? To be sure we commit everything that's already done before the loop.

Reply:
The only situation we don't commit everything to is when the timeout is small and the consumer exhausts that time on committing. That said, I think it's good to mark ready to commit -- it should be a cheap call and in most cases speed up `join` operations. Done in [d9c7666](https://github.com/getsentry/sentry/pull/40030/commits/d9c7666f843aded6cb1e387729f00a44bd3b6d4c).